### PR TITLE
Checkbox theme v2

### PR DESCRIPTION
**Pull request by OlenaKashuba**
Support background color
Custom icon

### CODE REVIEW

**Review by zopa-charlotte-walker — APPROVED** [review]
LGTM

File: src/components/molecules/CheckboxField/__snapshots__/CheckboxField.test.tsx.snap [hunk 1]
```diff
@@ -102,6 +102,7 @@ exports[`<CheckboxField /> renders the component with no control icon and no a11
 .c1:focus + label {
   border-color: #009E8E;
   box-shadow: 0 0 4px 0 #009E8E;
+  background-color: transparent;
```
Comment (admmasters) [inline src/components/molecules/CheckboxField/__snapshots__/CheckboxField.test.tsx.snap:105]: We happy that the default focus background is transparent now?

File: src/components/molecules/CheckboxField/__snapshots__/CheckboxField.test.tsx.snap [hunk 2]
```diff
@@ -110,6 +111,11 @@ exports[`<CheckboxField /> renders the component with no control icon and no a11
   box-shadow: 0 0 4px 0 #009E8E;
 }
 
+.c1:hover:checked:not(:disabled) + label:before,
+.c1:focus:checked + label:before {
+  background-color: #FFFFFF;
```
Comment (admmasters) [inline src/components/molecules/CheckboxField/__snapshots__/CheckboxField.test.tsx.snap:116]: And this one?

File: src/components/molecules/CheckboxField/__snapshots__/CheckboxField.test.tsx.snap [hunk 3]
```diff
@@ -129,13 +135,18 @@ exports[`<CheckboxField /> renders the component with no control icon and no a11
...
 
+.c1:disabled:before {
+  background-color: #FFFFFF;
+}
+
 .c1:checked + label {
   border-color: #009E8E;
   background-color: #ECFCF9;
-  box-shadow: none;
+  box-shadow: none #009E8E;
```
Comment (admmasters) [inline src/components/molecules/CheckboxField/__snapshots__/CheckboxField.test.tsx.snap:145]: This one seems like it'll have an effect on zopa 🤔 ?

**Review by admmasters — CHANGES_REQUESTED** [review]
Waiting for ok on the snapshots - if this is ok then happy to approve

File: src/components/molecules/CheckboxField/__snapshots__/CheckboxField.test.tsx.snap (hunk 3 above)
Comment (zopa-charlotte-walker) [reply src/components/molecules/CheckboxField/__snapshots__/CheckboxField.test.tsx.snap:145]: if its `none` + a colour, isnt that still none?

File: src/components/molecules/CheckboxField/__snapshots__/CheckboxField.test.tsx.snap (hunk 2 above)
Comment (zopa-charlotte-walker) [reply src/components/molecules/CheckboxField/__snapshots__/CheckboxField.test.tsx.snap:116]: does it make more sense to have either both transparent or both white so theyre consistent

File: src/components/molecules/CheckboxField/__snapshots__/CheckboxField.test.tsx.snap (hunk 3 above)
Comment (OlenaKashuba) [reply src/components/molecules/CheckboxField/__snapshots__/CheckboxField.test.tsx.snap:145]: yeah its still a none, its a combined string so to keep theme easier I left it like that. not ideal, I know...

**Review by admmasters — APPROVED** [review]

**Pull request merged**